### PR TITLE
update for min_chunk_bytes_for_parallel_parsing

### LINKED ISSUE #444: Small adjustment about INSERT Atomicity
Insert isn't atomic if `input_format_parallel_parsing` setting is enabled (and it's enabled by default)

https://github.com/ClickHouse/ClickHouse/issues/23303

https://clickhouse.com/docs/en/guides/developer/transactional

### PR DESCRIPTION
**Pull request by DanRoscigno**
closes https://github.com/ClickHouse/clickhouse-docs/issues/444

### Changelog category (leave one):
- Documentation (changelog entry is not required)

@UnamedRus @Avogar I split the doc up into cases as it made sense to me.  Does it make sense to have it this way to you?

If so, I think I need a case for when `max_insert_block_size` or `min_chunk_bytes_for_parallel_parsing` are exceeded.  If one or both are exceeded does that mean that each segment is atomic? (like Case 2 and Case 3)?

I am marking this as a draft to make sure that it does not get merged until we are ready, I could not add you as a reviewer @UnamedRus

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 88bf4e49d4767b6a3a3ccfcc383a42ca90ae12f1 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50452/88bf4e49d4767b6a3a3ccfcc383a42ca90ae12f1/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docs/en/sql-reference/transactions.md [hunk 1]
```diff
@@ -3,23 +3,44 @@ slug: /en/guides/developer/transactional
 ---
 # Transactional (ACID) support
 
-INSERT into one partition* in one table* of MergeTree* family up to max_insert_block_size rows* is transactional (ACID):
-- Atomic: INSERT is succeeded or rejected as a whole: if confirmation is sent to the client, all rows INSERTed; if error is sent to the client, no rows INSERTed.
+## Case 1: INSERT into one partition, of one table, of the MergeTree* family
+
+This is transactional (ACID) if the number of rows inserted is less than or equal to `max_insert_block_size rows`, and in the case of data in TSV, TKSV, CSV, or JSONEachRow format if the number of bytes is less than `min_chunk_bytes_for_parallel_parsing`:
```
Comment (Avogar) [inline docs/en/sql-reference/transactions.md:8]: 1) Not only `TSV, TKSV, CSV, or JSONEachRow` supports parallel parsing and setting `min_chunk_bytes_for_parallel_parsing`
2) `min_chunk_bytes_for_parallel_parsing` works only when parallel parsing is enabled
3) Setting `min_chunk_bytes_for_parallel_parsing` doesn't override setting `max_insert_block_size`. It tells parallel parser the size of chunk of raw data to parse in one thread. And after chunk is parsed, it's splitted into blocks using setting `max_insert_block_size`. So, I don't think we should mention this setting here at all

**Comment by Avogar** [comment]
> I think I need a case for when `max_insert_block_size` or `min_chunk_bytes_for_parallel_parsing` are exceeded

I am not sure we can exceede `max_insert_block_size`, this setting should limit the number of rows we insert. Maybe if it's too small we can squash blocks while inserting according to settings `min_insert_block_size_rows/min_insert_block_size_bytes`, so the result block will exceed `max_insert_block_size`, but not sure if it influese transactions.
Let's ask @tavplubix, he knows transactions much better than me.

**Comment by tavplubix** [comment]
It's always atomic with actual transactions (an experimental feature), settings for block size do not matter. And without transactions it's atomic only when you insert one block into one table. Unfortunately, we have too many settings that may affect block size on insertion (`max_block_size`, `max_insert_block_size`, `min_insert_block_size_rows`, `min_insert_block_size_bytes`, `min_insert_block_size_rows_for_materialized_views`, `min_insert_block_size_bytes_for_materialized_views`, `preferred_block_size_bytes`, `min_chunk_bytes_for_parallel_parsing`) and these settings interact with each other in a strange way, so I don't know how to set them to get only one block.

**Review by Avogar — APPROVED** [review]

**Comment by UnamedRus** [comment]
> and these settings interact with each other in a strange way, so I don't know how to set them to get only one block.

Not all of them, but for simple inserts it's enough to change those settings

https://kb.altinity.com/altinity-kb-queries-and-syntax/atomic-insert/#insert-with-adjusted-settings-atomic

**Pull request merged**